deployments/puppet/manifests/params.pp

```diff
@@ -26,7 +26,7 @@
     $auto_instrumentation_version = 'latest'
     $auto_instrumentation_java_agent_jar = '/usr/lib/splunk-instrumentation/splunk-otel-javaagent.jar'
 
-  } elsif $::osfamily == 'windows' {
+  } elsif $::osfamily == 'windows' or $::family == 'windows' {
```

Comment:
- `$::osfamily` is a legacy fact and should still be available to use in manifests, but are just hidden in the facter output by default.  From https://www.puppet.com/docs/puppet/8/core_facts.html:

  > Legacy Facts Note: As of Facter 3, legacy facts such as architecture are hidden by default to reduce noise in Facter's default command-line output. These older facts are now part of more useful structured facts; for example, architecture is now part of the os fact and accessible as os.architecture. You can still use these legacy facts in Puppet manifests ($architecture), request them on the command line (facter architecture), and view them alongside structured facts (facter --show-legacy). Legacy facts are excluded by default.

- I'm not sure if `$::family` is a valid fact. To use the non-legacy structured fact, the correct syntax should be `$facts['os']['family']` (see https://www.puppet.com/docs/puppet/8/lang_facts_accessing).  We can try changing all references to `$::osfamily` to `$facts['os']['family']` in our manifests to see if that helps.